src/smbc.rs

```diff
@@ -61,6 +62,16 @@ fn is_einval<T: Eq + From<i8>>(t: T) -> IoResult<T> {
     }
 }
 
+/// Converts a pointer to the beginning of a C string and it's length and puts its
+/// equivelant value casted to u8
+fn str_ptr_to_u8(str_len: u32, str_ptr: *const i8, out_buffer: &mut Vec<u8>) {
```

Comment:
I'm wondering if `Cstr::from_ptr()` would do the same thing as this?  I think it does without taking ownership.  Or a `CString::from_raw()` to take ownership.  Maybe I'm missing something here.

Reply:
Yeah, looking at this again we can just use CStr::from_ptr